PushTest.wsWithXhrBackgroundRequestAndSessionDoesNotActive failing on Flow 24.3

The test is failing often on Flow 24.3 nightly build

```
08:46:34   [ERROR] com.vaadin.cdi.itest.PushTest.wsWithXhrBackgroundRequestAndSessionDoesNotActive  Time elapsed: 4.043 s  <<< ERROR!
08:46:34   java.lang.RuntimeException: wsWithXhrBackgroundRequestAndSessionDoesNotActive(com.vaadin.cdi.itest.PushTest): run failed 2 times
08:46:34     at com.vaadin.testbench.RetryRule$1.evaluate(RetryRule.java:96)
08:46:34     at org.jboss.arquillian.junit.Arquillian$7$1.invoke(Arquillian.java:279)
08:46:34     at org.jboss.arquillian.container.test.impl.execution.ClientBeforeAfterLifecycleEventExecuter.execute(ClientBeforeAfterLifecycleEventExecuter.java:88)
08:46:34     at org.jboss.arquillian.container.test.impl.execution.ClientBeforeAfterLifecycleEventExecuter.on(ClientBeforeAfterLifecycleEventExecuter.java:66)
08:46:34     at jdk.internal.reflect.GeneratedMethodAccessor24.invoke(Unknown Source)
08:46:34     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
08:46:34     at java.base/java.lang.reflect.Method.invoke(Method.java:568)
08:46:34     at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
08:46:34     at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
08:46:34     at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
08:46:34     at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createContext(ContainerEventController.java:128)
08:46:34     at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createBeforeContext(ContainerEventController.java:114)
08:46:34     at jdk.internal.reflect.GeneratedMethodAccessor23.invoke(Unknown Source)
08:46:34     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
08:46:34     at java.base/java.lang.reflect.Method.invoke(Method.java:568)
08:46:34     at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
08:46:34     at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
08:46:34     at org.jboss.arquillian.test.impl.TestContextHandler.createTestContext(TestContextHandler.java:116)
08:46:34     at jdk.internal.reflect.GeneratedMethodAccessor22.invoke(Unknown Source)
08:46:34     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
08:46:34     at java.base/java.lang.reflect.Method.invoke(Method.java:568)
08:46:34     at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
08:46:34     at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
08:46:34     at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:83)
08:46:34     at jdk.internal.reflect.GeneratedMethodAccessor21.invoke(Unknown Source)
08:46:34     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
08:46:34     at java.base/java.lang.reflect.Method.invoke(Method.java:568)
08:46:34     at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
08:46:34     at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
08:46:34     at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)
08:46:34     at jdk.internal.reflect.GeneratedMethodAccessor20.invoke(Unknown Source)
08:46:34     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
08:46:34     at java.base/java.lang.reflect.Method.invoke(Method.java:568)
08:46:34     at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
08:46:34     at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
08:46:34     at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
08:46:34     at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
08:46:34     at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.fireCustomLifecycle(EventTestRunnerAdaptor.java:159)
08:46:34     at org.jboss.arquillian.junit.Arquillian$7.evaluate(Arquillian.java:273)
08:46:34     at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
08:46:34     at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
08:46:34     at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
08:46:34     at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
08:46:34     at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
08:46:34     at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
08:46:34     at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
08:46:34     at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
08:46:34     at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
08:46:34     at org.jboss.arquillian.junit.Arquillian$2.evaluate(Arquillian.java:166)
08:46:34     at org.jboss.arquillian.junit.Arquillian.multiExecute(Arquillian.java:350)
08:46:34     at org.jboss.arquillian.junit.Arquillian.access$200(Arquillian.java:54)
08:46:34     at org.jboss.arquillian.junit.Arquillian$3.evaluate(Arquillian.java:177)
08:46:34     at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
08:46:34     at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
08:46:34     at org.jboss.arquillian.junit.Arquillian.run(Arquillian.java:115)
08:46:34     at org.junit.runners.Suite.runChild(Suite.java:128)
08:46:34     at org.junit.runners.Suite.runChild(Suite.java:27)
08:46:34     at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
08:46:34     at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
08:46:34     at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
08:46:34     at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
08:46:34     at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
08:46:34     at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
08:46:34     at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
08:46:34     at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
08:46:34     at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
08:46:34     at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
08:46:34     at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
08:46:34     at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
08:46:34     at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
08:46:34     at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
08:46:34     at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
08:46:34     at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
08:46:34     at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
08:46:34   Caused by: org.junit.ComparisonFailure: expected:<[fals]e> but was:<[tru]e>
08:46:34     at org.junit.Assert.assertEquals(Assert.java:117)
08:46:34     at org.junit.Assert.assertEquals(Assert.java:146)
08:46:34     at com.vaadin.cdi.itest.AbstractCdiTest.assertTextEquals(AbstractCdiTest.java:81)
08:46:34     at com.vaadin.cdi.itest.PushTest.assertContextActive(PushTest.java:94)
08:46:34     at com.vaadin.cdi.itest.PushTest.assertAllExceptRequestAndSessionActive(PushTest.java:79)
08:46:34     at com.vaadin.cdi.itest.PushTest.wsWithXhrBackgroundRequestAndSessionDoesNotActive(PushTest.java:50)
08:46:34     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
08:46:34     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
08:46:34     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
08:46:34     at java.base/java.lang.reflect.Method.invoke(Method.java:568)
08:46:34     at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
08:46:34     at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
08:46:34     at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
08:46:34     at org.jboss.arquillian.junit.Arquillian$8$1.invokeMethod(Arquillian.java:325)
08:46:34     at org.jboss.arquillian.junit.MethodInvoker$1.invoke(MethodInvoker.java:18)
08:46:34     at org.jboss.arquillian.container.test.impl.execution.LocalTestExecuter.execute(LocalTestExecuter.java:57)
08:46:34     at jdk.internal.reflect.GeneratedMethodAccessor56.invoke(Unknown Source)
08:46:34     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
08:46:34     at java.base/java.lang.reflect.Method.invoke(Method.java:568)
08:46:34     at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
08:46:34     at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
08:46:34     at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
08:46:34     at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
08:46:34     at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
08:46:34     at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
08:46:34     at org.jboss.arquillian.container.test.impl.execution.ClientTestExecuter.execute(ClientTestExecuter.java:50)
08:46:34     at jdk.internal.reflect.GeneratedMethodAccessor55.invoke(Unknown Source)
08:46:34     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
08:46:34     at java.base/java.lang.reflect.Method.invoke(Method.java:568)
08:46:34     at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
08:46:34     at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
08:46:34     at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
08:46:34     at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createContext(ContainerEventController.java:128)
08:46:34     at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createTestContext(ContainerEventController.java:118)
08:46:34     at jdk.internal.reflect.GeneratedMethodAccessor54.invoke(Unknown Source)
08:46:34     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
08:46:34     at java.base/java.lang.reflect.Method.invoke(Method.java:568)
08:46:34     at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
08:46:34     at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
08:46:34     at org.jboss.arquillian.test.impl.TestContextHandler.createTestContext(TestContextHandler.java:116)
08:46:34     at jdk.internal.reflect.GeneratedMethodAccessor22.invoke(Unknown Source)
08:46:34     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
08:46:34     at java.base/java.lang.reflect.Method.invoke(Method.java:568)
08:46:34     at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
08:46:34     at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
08:46:34     at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:83)
08:46:34     at jdk.internal.reflect.GeneratedMethodAccessor21.invoke(Unknown Source)
08:46:34     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
08:46:34     at java.base/java.lang.reflect.Method.invoke(Method.java:568)
08:46:34     at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
08:46:34     at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
08:46:34     at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)
08:46:34     at jdk.internal.reflect.GeneratedMethodAccessor20.invoke(Unknown Source)
08:46:34     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
08:46:34     at java.base/java.lang.reflect.Method.invoke(Method.java:568)
08:46:34     at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
08:46:34     at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
08:46:34     at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
08:46:34     at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.test(EventTestRunnerAdaptor.java:139)
08:46:34     at org.jboss.arquillian.junit.MethodInvoker.invoke(MethodInvoker.java:15)
08:46:34     at org.jboss.arquillian.junit.Arquillian$8.evaluate(Arquillian.java:332)
08:46:34     at org.jboss.arquillian.junit.Arquillian$4.evaluate(Arquillian.java:204)
08:46:34     at org.jboss.arquillian.junit.Arquillian.multiExecute(Arquillian.java:350)
08:46:34     at org.jboss.arquillian.junit.Arquillian.access$200(Arquillian.java:54)
08:46:34     at org.jboss.arquillian.junit.Arquillian$5.evaluate(Arquillian.java:215)
08:46:34     at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
08:46:34     at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
08:46:34     at com.vaadin.testbench.RetryRule$1.evaluate(RetryRule.java:73)
08:46:34     ... 73 more
08:46:34   
```

--- a/vaadin-cdi-itest/src/main/java/com/vaadin/cdi/itest/push/PushComponent.java
+++ b/vaadin-cdi-itest/src/main/java/com/vaadin/cdi/itest/push/PushComponent.java
@@ -16,15 +16,17 @@
 
 package com.vaadin.cdi.itest.push;
 
-import java.lang.annotation.Annotation;
-import java.util.concurrent.locks.Lock;
-
 import jakarta.annotation.PostConstruct;
 import jakarta.annotation.Resource;
+import jakarta.enterprise.concurrent.ManagedScheduledExecutorService;
 import jakarta.enterprise.concurrent.ManagedThreadFactory;
 import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.enterprise.context.RequestScoped;
 import jakarta.enterprise.context.SessionScoped;
+import java.lang.annotation.Annotation;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
 
 import com.vaadin.cdi.annotation.CdiComponent;
 import com.vaadin.cdi.annotation.RouteScoped;
@@ -36,6 +38,7 @@ import com.vaadin.flow.component.UI;
 import com.vaadin.flow.component.html.Div;
 import com.vaadin.flow.component.html.Label;
 import com.vaadin.flow.component.html.NativeButton;
+import com.vaadin.flow.component.html.Span;
 
 @CdiComponent
 public class PushComponent extends Div {
@@ -55,20 +58,24 @@ public class PushComponent extends Div {
 
         @Override
         public void run() {
-            // We can acquire the lock after the request started this thread is processed
-            // Needed to make sure that this is sent as a push message
+            ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
             Thread.currentThread().setContextClassLoader(this.classLoader);
-            Lock lockInstance = ui.getSession().getLockInstance();
-            lockInstance.lock();
-            lockInstance.unlock();
+            try {
+                // We can acquire the lock after the request started this thread is processed
+                // Needed to make sure that this is sent as a push message
+                Lock lockInstance = ui.getSession().getLockInstance();
+                lockInstance.lock();
+                lockInstance.unlock();
 
-            ui.access(PushComponent.this::print);
+                ui.access(PushComponent.this::print);
+            } finally {
+                Thread.currentThread().setContextClassLoader(contextClassLoader);
+            }
         }
-
     }
 
     @Resource
-    private ManagedThreadFactory threadFactory;
+    private ManagedScheduledExecutorService executorService;
 
     private void print() {
         printContextIsActive(RequestScoped.class);
@@ -90,8 +97,9 @@ public class PushComponent extends Div {
     private void init() {
         NativeButton bgButton = new NativeButton("background", event -> {
             ContextCheckTask task = new ContextCheckTask(UI.getCurrent());
-            Thread thread = threadFactory.newThread(task);
-            thread.start();
+            // Delay execution to prevent UIDL being written on current
+            // request instead of being pushed through websocket
+            executorService.schedule(task, 10, TimeUnit.MILLISECONDS);
         });
         bgButton.setId(RUN_BACKGROUND);
 

--- a/vaadin-cdi-itest/src/main/java/com/vaadin/cdi/itest/push/PushComponent.java
+++ b/vaadin-cdi-itest/src/main/java/com/vaadin/cdi/itest/push/PushComponent.java
@@ -19,14 +19,12 @@ package com.vaadin.cdi.itest.push;
 import jakarta.annotation.PostConstruct;
 import jakarta.annotation.Resource;
 import jakarta.enterprise.concurrent.ManagedScheduledExecutorService;
-import jakarta.enterprise.concurrent.ManagedThreadFactory;
 import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.enterprise.context.RequestScoped;
 import jakarta.enterprise.context.SessionScoped;
 import java.lang.annotation.Annotation;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.locks.Lock;
-import java.util.concurrent.locks.ReentrantLock;
 
 import com.vaadin.cdi.annotation.CdiComponent;
 import com.vaadin.cdi.annotation.RouteScoped;
@@ -38,7 +36,6 @@ import com.vaadin.flow.component.UI;
 import com.vaadin.flow.component.html.Div;
 import com.vaadin.flow.component.html.Label;
 import com.vaadin.flow.component.html.NativeButton;
-import com.vaadin.flow.component.html.Span;
 
 @CdiComponent
 public class PushComponent extends Div {
@@ -97,8 +94,8 @@ public class PushComponent extends Div {
     private void init() {
         NativeButton bgButton = new NativeButton("background", event -> {
             ContextCheckTask task = new ContextCheckTask(UI.getCurrent());
-            // Delay execution to prevent UIDL being written on current
-            // request instead of being pushed through websocket
+            // Delay execution to prevent UIDL being pushed by the thread service
+            // the current HTTP request instead of the background thread
             executorService.schedule(task, 10, TimeUnit.MILLISECONDS);
         });
         bgButton.setId(RUN_BACKGROUND);

--- a/vaadin-cdi-itest/src/main/java/com/vaadin/cdi/itest/push/PushComponent.java
+++ b/vaadin-cdi-itest/src/main/java/com/vaadin/cdi/itest/push/PushComponent.java
@@ -94,7 +94,7 @@ public class PushComponent extends Div {
     private void init() {
         NativeButton bgButton = new NativeButton("background", event -> {
             ContextCheckTask task = new ContextCheckTask(UI.getCurrent());
-            // Delay execution to prevent UIDL being pushed by the thread service
+            // Delay execution to prevent UIDL being pushed by the thread serving
             // the current HTTP request instead of the background thread
             executorService.schedule(task, 10, TimeUnit.MILLISECONDS);
         });

